pkg/ccl/backupccl/backup_test.go

```diff
@@ -8703,3 +8703,25 @@ DROP TABLE foo;
 	defer cleanupEmptyCluster()
 	sqlDBRestore.Exec(t, "RESTORE FROM $1 AS OF SYSTEM TIME "+aost, LocalFoo)
 }
+
+func TestRestoreNewDatabaseName(t *testing.T) {
```

Comment:
In the test below let's also create a table in the db, and insert a few rows. Post restore lets `SELECT count(*)` on the table and verify that we see the same number of rows.

Some more tests I think we should add, probably as subtests (see t.Run() method in other tests):

- Option being incorrectly used in full cluster restore, table restore, multiple target DB restore.

Reply:
done.